### DISCUSSION TITLE
How to use my own DisplayDriver<User> while still using OrchardCore.Users Module.

### DISCUSSION THREAD

**Discussion by XpandIT-RFID** [OrchardCMS/OrchardCore#10108] [Q&A] [answered]
HI, so I have some "custom settings" in terms of the field of the identity of the users, I had to "create" my own custom UsersRegistration Module.
Now I have a problem when it comes to the verification of emails. The users display driver calls a shape named "UserButtons" 
![image](https://user-images.githubusercontent.com/83280308/129380836-7a2e29ae-8dd1-43f0-aef8-bc8deda81873.png)
Which contains this button
![image](https://user-images.githubusercontent.com/83280308/129380887-03020176-76c7-4e79-8ec3-a20b5eb9d63c.png)
And it won't work unless it has that "asp-area" parameter in the element since I am using my own Registration Controller.
I really need a way to either override the UserButtons.cshtml or the implementation of IDisplayDriver<IUser>.

EDIT: Also in the future I will be needing to show a UserCustomSettings Field Value instead of the username in the list so it would be easier if there was a way to replace the implementation of IDisplayDriver<IUser>. I already tried to replace the scoped in the startup but had no luck, it used both.

Any suggestions? thanks!

**Comment by ns8482e** [comment, marked as answer]
If you need it replace the user button with your own shape,  easiest way to hide the UserButtons using placement and define your own shape using your own driver for User in your module

**Reply by XpandIT-RFID** [reply]
Hi @ns8482e, could you please give me an example of how to do the placement part? I still don't get how that works

**Comment by XpandIT-RFID** [comment]
Got it, thanks to @ns8482e :)
All I had to do was create the shapes in my module an add the placements.json in the root of the module with this json:

```
{
  "UserButtons": [
    {
      "shape": "MyUserButtons"
    }
  ],
  "UserFields": [
    {
      "shape": "MyUserFields"
    }
  ]
}

```
I feel like this needs better documentation, it has a lot of optional fields and can be a little tricky to understand. Still another cool trick in OrchardCore that I learned today :)
